contracts/multisig-prover/src/state.rs

```diff
@@ -24,3 +28,42 @@ pub const COMMANDS_BATCH: Map<&BatchID, CommandBatch> = Map::new("command_batch"
 pub const MULTISIG_SESSION_BATCH: Map<u64, BatchID> = Map::new("multisig_session_batch");
 
 pub const REPLY_BATCH: Item<BatchID> = Item::new("reply_tracker");
+
+#[cw_serde]
+pub struct WorkerSet {
+    pub signers: Vec<Signer>,
+    pub threshold: Uint256,
+    pub id: u64, // for randomness
+}
+
+impl WorkerSet {
+    pub fn new(
+        snapshot: Snapshot,
+        pub_keys: Vec<multisig::types::PublicKey>,
+        env: Env,
```

Comment:
pass in the block height instead

Reply:
Agree with @cgorenflo. This goes again with my comment saying that I don't like passing exessive information to any function.